.github/workflows/cost-estimate.yml

```diff
@@ -0,0 +1,335 @@
+name: Cost Estimate Test
+
+permissions:
+  contents: read
+  pull-requests: write
+  checks: write
+
+on:
+  pull_request:
+    paths:
+      - 'soroban-sdk/**'
+      - '.github/workflows/cost-estimate.yml'
+  push:
+    branches: [main]
+    paths:
+      - 'soroban-sdk/**'
+      - '.github/workflows/cost-estimate.yml'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  RUSTFLAGS: -C debuginfo=0 # Do not produce debug symbols to keep memory usage down
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+
+jobs:
+  cost-estimate:
+    name: Cost Estimate Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain: [stable]
+      fail-fast: false
+
+    steps:
+      # === Setup ===
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          components: rustfmt, clippy
+          target: wasm32-unknown-unknown
+
+      - name: Install expect-test dependencies
+        run: |
+          # Install expect-test if not already in dependencies
+          # This assumes expect-test is already configured in the project
+          echo "Expect-test should already be configured in Cargo.toml"
+
+      # === Run Cost Estimate Tests ===
+      - name: Run cost estimate tests
+        id: cost-test
+        run: |
+          echo "## Running Cost Estimate Tests" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          
+          # Set environment variables for structured output
+          export RUST_LOG=debug
+          export EXPECT_TEST=overwrite  # This will fail if snapshots drift
+          
+          # Create output directory
+          mkdir -p cost-estimate-output
+          
+          # Run the specific cost estimate test with detailed output
+          echo "Running cost estimate tests on ${{ matrix.toolchain }} toolchain..."
+          
+          # Capture both stdout and stderr, and test exit code
+          set +e
+          cargo test --package soroban-sdk cost_estimate --verbose -- --nocapture 2>&1 | tee cost-estimate-output/test-output-${{ matrix.toolchain }}.log
```

Comment:
Can be simplified to just calling `cargo test > ...` I think, like in the other workflow for the test themselves. In the end I might have a dedicated test_cost.rs or something like that.